### PR TITLE
Remove repeat test for types in test nn

### LINKED ISSUE #69865: Obliviate repeat_test_for_types usage in pytorch
### 🚀 The feature, motivation and pitch

Currently there are still a few tests in the codebase (mostly in test_nn.py) that use repeat_test_for_types to repeat a test for different data types. There is now a better way to do this with OpInfos which is now used across most of pytorch. We should migrate these tests away from using repeat_test_for_types and remove repeat_test_for_types in common_utils.py. 

### Alternatives

_No response_

### Additional context

The current test cases that use this decorator still are below.
In test_nn.py
* test_Conv2d_deterministic_cudnn
* test_Conv2d_large_workspace
* test_ConvTranspose2d_large_output_padding
* test_Conv2d_depthwise_naive_groups_cuda
* test_Conv3d_depthwise_naive_groups_cuda
* test_noncontig_conv_grad_cuda
* test_batchnorm_large_batch
* test_conv_double_backward_cuda

In test_data_parallel.py
* test_data_parallel_module
* test_data_parallel_module_kwargs_only
* test_data_parallel_module_kwargs_only_empty_list
* test_data_parallel_module_kwargs_only_empty_dict
* test_data_parallel_module_kwargs_only_empty_tuple

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @SciPioneer @H-Huang @albanD @mruberry @jbschlosser @walterddr @kshitij12345

### PR DESCRIPTION
**Pull request by janeyx99**
Helps fix a part of #69865

The first commit just migrates everything as is. 

The second commit uses the "device" variable instead of passing "cuda" everywhere

### CODE REVIEW

File: test/test_nn.py [hunk 1]
```diff
@@ -11165,28 +11004,6 @@ def test_conv_cudnn_memory_layout_dominance(self):
         out = conv(input)
         self.assertTrue(out.is_contiguous(memory_format=torch.channels_last))
 
-    def test_conv_double_backward(self):
```
Comment (janeyx99) [inline test/test_nn.py:11168]: Deleted this since i removed the cuda suffix of the migrated test_conv_double_backward_cuda

File: test/test_nn.py [hunk 2]
```diff
@@ -13243,6 +13037,204 @@ def test_affine_3d_rotateRandom(self, device):
...
+                             torch.cat([m1.bias.grad.data,
+                                        m2.bias.grad.data], 0),
+                             atol=dtype2prec_DONTUSE[dtype], rtol=0)
+            self.assertEqual(m.weight.grad.data,
+                             torch.cat([m1.weight.grad.data,
+                                        m2.weight.grad.data], 0),
+                             atol=dtype2prec_DONTUSE[dtype], rtol=0)
+
+    @onlyCUDA
+    @dtypes(*ALL_TENSORTYPES)
```
Comment (jbschlosser) [inline test/test_nn.py:13145]: Can the instances of `ALL_TENSORTYPES` be replaced with the new improved way of specifying these? cc @mruberry

**Review by jbschlosser — COMMENTED** [review]
Awesome, thanks for the fast cleanup! Looks pretty solid overall :)

I went through all the DISABLED test issues and it doesn't look like any need to be renamed (although you might want to check my work).

I'll let @mruberry make a pass but otherwise it's good to go on my end.

File: test/test_nn.py (hunk 2 above)
Comment (janeyx99) [reply test/test_nn.py:13145]: What is the new improved way?

Comment (jbschlosser) [reply test/test_nn.py:13145]: I'm not exactly sure, but Mike should know

File: test/test_nn.py [hunk 3]
```diff
@@ -13243,6 +13037,204 @@ def test_affine_3d_rotateRandom(self, device):
...
+    @onlyCUDA
+    @dtypes(*DOUBLE_TENSORTYPES)
+    def test_conv_double_backward(self, device, dtype):
+        with torch.backends.cudnn.flags(deterministic=True):
+            # Double backward only runs with DoubleTensor due to precision reason
+            batch_size = 1
+            for kern, inp_size, dilations in [(3, 5, [1, 2]), (4, 9, [1])]:
+                for stride, padding, chan_in, chan_out, dilation in product([1], [2], [2], [3], dilations):
+                    no_weight = stride == 2
+                    result = self.run_conv_double_back_test(kern, stride,
```
Comment (jbschlosser) [inline test/test_nn.py:13222]: Based on the failing check, it looks like this helper function needs to be brought over to `TestNNDeviceType` from `TestNN` as well

**Review by jbschlosser — APPROVED** [review]
LGTM

File: test/test_nn.py (hunk 2 above)
Comment (mruberry) [reply test/test_nn.py:13081]: I would just list the dtypes explicitly. This test historically ran in double, float, and half:

https://github.com/pytorch/pytorch/blob/2367face24afb159f73ebf40dc6f23e46132b770/torch/testing/_internal/common_utils.py#L813

`ALL_TENSORTYPES` is highly misleading and creating a list of dtypes like this isn't a great idea. We do have dtype lists in our test suite but they correspond with the ATen dispatch macros and are available through https://github.com/pytorch/pytorch/blob/master/torch/testing/_internal/common_dtype.py. They are actually defined here:

https://github.com/pytorch/pytorch/blob/10b55648f5deb9a2e9d4c6b7f75836da2b96c6bd/torch/testing/_legacy.py#L63

This particular list corresponds to `floating_types_and_half`, but using these isn't necessarily preferable for these tests. Those dtype lists are intended to be used with {Op, Module}Infos where the correspondence with the ATen dispatch macros is closer.

File: test/test_nn.py [hunk 4]
```diff
@@ -13243,6 +12973,270 @@ def test_affine_3d_rotateRandom(self, device):
 
             self.assertEqual(scipy_ary, gridsample_ary.reshape_as(scipy_ary))
 
+
+    @onlyCUDA
+    @skipCUDAIfNoCudnn
+    @dtypes(*get_all_fp_dtypes(include_bfloat16=AMPERE_OR_ROCM))
```
Comment (mruberry) [inline test/test_nn.py:12979]: Would be nice to kill `get_all_fp_dtypes` usage here, too

File: test/test_nn.py [hunk 5]
```diff
@@ -12725,8 +12411,52 @@ def _buildEquivalentAffineTransforms3d(device, input_size, output_size, angle_ra
 
 
 class TestNNDeviceType(NNTestCase):
-    def _test_dropout(self, cls, device, input, memory_format=torch.contiguous_format):
-        p = 0.2
+    def run_conv_double_back_test(self, kern, stride, padding, chan_in, chan_out, batch_size,
+                                  inp_size, dilation, no_weight, groups=1, use_cuda=False,
+                                  use_bias=True, dtype=torch.double):
+        if use_cuda:
```
Comment (mruberry) [inline test/test_nn.py:12417]: This is OK but ideally the exact CUDA device passed to the test would be used. This test will always use the default CUDA device.

File: test/test_nn.py [hunk 6]
```diff
@@ -13243,6 +12973,270 @@ def test_affine_3d_rotateRandom(self, device):
...
+            self.assertEqual(out1, out2, atol=0.0, rtol=0)
+            y = torch.randn(out1.size(), device=device, dtype=dtype)
+            out1.backward(y)
+            out2.backward(y)
+            self.assertEqual(conv1.bias.grad.data, conv2.bias.grad.data, atol=0.0, rtol=0)
+            self.assertEqual(conv1.weight.grad.data, conv2.weight.grad.data, atol=0.0, rtol=0)
+
+
+    @onlyCUDA
+    @dtypes(*get_all_fp_dtypes(include_bfloat16=AMPERE_OR_ROCM))
```
Comment (mruberry) [inline test/test_nn.py:12998]: Same in that it'd be nice to kill the legacy dtype getter here

File: test/test_nn.py [hunk 7]
```diff
@@ -13243,6 +12973,270 @@ def test_affine_3d_rotateRandom(self, device):
...
+                                        m2.bias.grad.data], 0),
+                             atol=dtype2prec_DONTUSE[dtype], rtol=0)
+            self.assertEqual(m.weight.grad.data,
+                             torch.cat([m1.weight.grad.data,
+                                        m2.weight.grad.data], 0),
+                             atol=dtype2prec_DONTUSE[dtype], rtol=0)
+
+
+    @onlyCUDA
+    @dtypes(*get_all_fp_dtypes(include_bfloat16=AMPERE_OR_ROCM))
```
Comment (mruberry) [inline test/test_nn.py:13123]: TODO: update this

File: test/test_nn.py [hunk 8]
```diff
@@ -13243,6 +12973,270 @@ def test_affine_3d_rotateRandom(self, device):
...
+        bn = nn.BatchNorm2d(1).to(device, dtype)
+        data = torch.rand(880801, 1, 1, 1, device=device, dtype=dtype)
+        out = bn(data).sum().backward()
+
+
+    @onlyCUDA
+    @dtypes(torch.double)
+    def test_conv_double_backward(self, device, dtype):
+        with torch.backends.cudnn.flags(deterministic=True):
+            # Double backward only runs with DoubleTensor due to precision reason
```
Comment (mruberry) [inline test/test_nn.py:13153]: This comment looks outdated and can be removed

**Pull request closed without merging**